pkg/models/datasource_cache.go

```diff
@@ -72,7 +73,14 @@ type dataSourceTransport struct {
 
 func instrumentRoundtrip(datasourceName string, next http.RoundTripper) promhttp.RoundTripperFunc {
 	return promhttp.RoundTripperFunc(func(r *http.Request) (*http.Response, error) {
-		datasourceLabel := prometheus.Labels{"datasource": datasourceName}
+		datasourceLabelName, err := metrics.SanitizeLabelName(datasourceName)
```

Comment:
Maybe we should lowercase this label?

Reply:
Uppercase is allowed in Prometheus https://github.com/prometheus/common/blob/master/model/labels.go#L86

Lowercasing would increase the chance of collision so I would prefer to not do that.